### ISSUE TITLE
GraphicsContextException: Invalid window

### ISSUE THREAD

**Issue by elder-george** [AdamsLair/duality#155]
[Edit] ERROR:   An error occurred: GraphicsContextException at Sdl2GraphicsContext.Sdl2GraphicsContext(GraphicsMode, IWindowInfo, IGraphicsContext, Int32, Int32, GraphicsContextFlags): Invalid window
[Edit]          CallStack:
[Edit]             at OpenTK.Platform.SDL2.Sdl2GraphicsContext..ctor(GraphicsMode mode, IWindowInfo win, IGraphicsContext shareContext, Int32 major, Int32 minor, GraphicsContextFlags flags) in c:\Stuff\svn\Projects\AdamsLair.OpenTK\Source\OpenTK\Platform\SDL2\Sdl2GraphicsContext.cs:line 84
[Edit]             at OpenTK.Platform.SDL2.Sdl2Factory.CreateGLContext(GraphicsMode mode, IWindowInfo window, IGraphicsContext shareContext, Boolean directRendering, Int32 major, Int32 minor, GraphicsContextFlags flags) in c:\Stuff\svn\Projects\AdamsLair.OpenTK\Source\OpenTK\Platform\SDL2\Sdl2Factory.cs:line 70
[Edit]             at OpenTK.Graphics.GraphicsContext..ctor(GraphicsMode mode, IWindowInfo window, Int32 major, Int32 minor, GraphicsContextFlags flags) in c:\Stuff\svn\Projects\AdamsLair.OpenTK\Source\OpenTK\Graphics\GraphicsContext.cs:line 141
[Edit]             at OpenTK.Sdl2GLControl.CreateContext(Int32 major, Int32 minor, GraphicsContextFlags flags)
[Edit]             at OpenTK.GLControl.OnHandleCreated(EventArgs e)
[Edit]             at System.Windows.Forms.Control.WmCreate(Message& m)
[Edit]             at System.Windows.Forms.Control.WndProc(Message& m)
[Edit]             at System.Windows.Forms.ScrollableControl.WndProc(Message& m)
[Edit]             at System.Windows.Forms.UserControl.WndProc(Message& m)
[Edit]             at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
[Edit]             at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
[Edit]             at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

**Comment by elder-george** [comment]
In Windows Log there's one more error

Application: DualityEditor.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.Reflection.TargetInvocationException
Stack:
   at System.RuntimeMethodHandle.InvokeMethod(System.Object, System.Object[], System.Signature, Boolean)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(System.Object, System.Object[], System.Object[])
   at System.Delegate.DynamicInvokeImpl(System.Object[])
   at System.Windows.Forms.Control.InvokeMarshaledCallbackDo(ThreadMethodEntry)
   at System.Windows.Forms.Control.InvokeMarshaledCallbackHelper(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Windows.Forms.Control.InvokeMarshaledCallback(ThreadMethodEntry)
   at System.Windows.Forms.Control.InvokeMarshaledCallbacks()
   at System.Windows.Forms.Control.WndProc(System.Windows.Forms.Message ByRef)
   at System.Windows.Forms.ScrollableControl.WndProc(System.Windows.Forms.Message ByRef)
   at System.Windows.Forms.Form.WndProc(System.Windows.Forms.Message ByRef)
   at System.Windows.Forms.Control+ControlNativeWindow.OnMessage(System.Windows.Forms.Message ByRef)
   at System.Windows.Forms.Control+ControlNativeWindow.WndProc(System.Windows.Forms.Message ByRef)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr, Int32, IntPtr, IntPtr)
   at System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(MSG ByRef)
   at System.Windows.Forms.Application+ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr, Int32, Int32)
   at System.Windows.Forms.Application+ThreadContext.RunMessageLoopInner(Int32, System.Windows.Forms.ApplicationContext)
   at System.Windows.Forms.Application+ThreadContext.RunMessageLoop(Int32, System.Windows.Forms.ApplicationContext)
   at System.Windows.Forms.Application.Run()
   at Duality.Editor.Program.Main(System.String[])

**Comment by AdamsLair** [comment]
Hi elder-george,

as far as I can tell from looking at the callstack, there seems to be a problem in OpenTK window management. I'm a bit confused seeing the SDL2 backend there as well, because Duality doesn't use SDL2 by default and can only be made using it by explicitly making changes. Since they've switched to using the SDL2 backend in their local Duality fork, maybe @BraveSirAndrew or @Andrea know something about this?

Aside from this, can you provide the full logfile so we can see what exactly is going on there? How did you reproduce this error?

**Comment by BraveSirAndrew** [comment]
Hi @elder-george, not sure if it's any help, but all that needs to happen for the SDL backend to try and kick in is for you to have the SDL2.dll file placed in your game folder. If you do then the editor won't be able to start because GLControl, which Dualitor uses for WinForms integration, doesn't work at all with SDL2. Spent a week pulling my hair out over that and couldn't get it to work, and brighter minds have tried and failed to, so know what you're in for if you try and get it working:) The launcher should work with the sdl library present though. Can't remember if we had to do much tinkering there or not, but I don't think we did.
